app.js

```diff
@@ -92,26 +94,31 @@ function createRandomProducts() {
 
 function generateUniqueIndex() {
   let uniqueIndexNumbers = [];
+
   while (uniqueIndexNumbers.length < 3) {
     let randomNumber = createRandomProducts();
-    if (!uniqueIndexNumbers.includes(randomNumber)) {
+    if (!uniqueIndexNumbers.includes(randomNumber) && !priorIndexNumbers.includes(randomNumber)) {
       uniqueIndexNumbers.push(randomNumber);
+      priorIndexNumbers.push(randomNumber);
     }
   }
+  if (priorIndexNumbers.length === 6){
+    priorIndexNumbers.splice(0,3);
+  }
   return uniqueIndexNumbers;
```

Comment:
Great job! My only comment is that this doesn't scale well. If you ever want to show more than three images, you're going to have to change all the numbers here.